cnd/src/db/mod.rs

```diff
@@ -107,35 +141,37 @@ fn create_database_if_needed(db: &Path) -> Result<(), Error> {
         log::info!("Creating Sqlite database: {}", db.display());
 
         if let Some(parent) = db.parent() {
-            std::fs::create_dir_all(parent).map_err(|err| Error::Path(err.to_string()))?;
+            std::fs::create_dir_all(parent)?;
         }
 
-        File::create(db).map_err(|err| Error::Path(err.to_string()))?;
+        File::create(db)?;
     }
     Ok(())
 }
 
 fn establish_connection(db: &Path) -> Result<SqliteConnection, Error> {
-    let database_url = db
-        .to_str()
-        .ok_or_else(|| Error::Path(format!("Database path invalid: {}", db.to_string_lossy())))?;
-
-    SqliteConnection::establish(&database_url).map_err(|err| Error::Connect(err.to_string()))
+    let database_url = db.to_str().ok_or(Error::PathStr)?;
+    let conn = SqliteConnection::establish(&database_url)?;
+    Ok(conn)
 }
 
-impl TryFrom<Metadata> for metadata_store::Metadata {
-    type Error = metadata_store::Error;
+impl TryFrom<Metadata> for ms::Metadata {
```

Comment:
:upside_down_face: Could be clearer.
```suggestion
impl TryFrom<db::models::Metadata> for metadata_store::Metadata {
```
Not too sure about using an acronym, but it's a minor thing.

Reply:
Discussed offline; aliasing (acronym) for package makes the code clearer here because it makes the code less cluttered without introducing unnecessary extra cognitive load on the developer.  This is different to aliasing types which causes grief in typical IDE workflow.  Thanks.